internal/provider/resource_port_test.go

```diff
@@ -2,65 +2,57 @@ package provider
 
 import (
 	"fmt"
+	"os"
 	"testing"
 
 	"github.com/PacketFabric/terraform-provider-packetfabric/internal/testutil"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 )
 
-func hclPort(description, accountUUID, speed, media, subscriptionTerm, pop, zone string, autoneg, nni bool) (hcl string, resourceName string) {
+func hclPort(description, media, speed, pop, subscriptionTerm string) (hcl string, resourceName string) {
 	hclName := testutil.GenerateUniqueResourceName()
 	resourceName = "packetfabric_port." + hclName
-	hcl = fmt.Sprintf(`
-	resource "packetfabric_port" "%s" {
-		description       = "%s"
-		account_uuid      = "%s"
-		speed             = "%s"
-		media             = "%s"
-		subscription_term = "%s"
-		pop               = "%s"
-		zone              = "%s"
-		autoneg           = %t
-		nni               = %t
-	}`, hclName, description, accountUUID, speed, media, subscriptionTerm, pop, zone, autoneg, nni)
+	hcl = fmt.Sprintf(testutil.RResourcePort, hclName, description, media, pop, speed, subscriptionTerm)
 	return
 }
 
 func TestAccPort(t *testing.T) {
 	testutil.SkipIfEnvNotSet(t)
 
-	description := testutil.GenerateUniqueName(testPrefix)
-	pop, zone, err := testutil.GetPopAndZoneWithAvailablePort("1Gbps", "LX")
-	if err != nil {
-		t.Fatalf("Unable to find pop and zone with available port: %s", err)
-	}
+	description := os.Getenv(testutil.PF_PORT_DESCR_KEY)
+	media := os.Getenv(testutil.PF_PORT_MEDIA_KEY)
+	speed := os.Getenv(testutil.PF_PORT_SPEED_KEY)
+	pop := os.Getenv(testutil.PF_PORT_POP1_KEY)
```

Comment:
Pop/zone should be selected automatically in the test code. It would be terrible UX to run tests if we have to provide this data.

Reply:
Could you please share the API that must be called to get the Pop/zone dynamically? Thanks.

Reply:
I agree with @medzin, this requirement was discussed in the kickoff meeting.

The existing ACC in the repo was already doing that. It sounds like the logic is [here].(https://github.com/PacketFabric/terraform-provider-packetfabric/blob/809b44b031a878b573741ebc602b020b98ad134b/internal/testutil/testutil.go#L31)

Reply:
Great, we gonna update the PR as well.

Reply:
We solved this issue with the latest changes.